File: PYTHON/array-leader.py
# arrSize=int(input())
# A=[]
# for i in range(arrSize):
#     A.append(int(input()))
# print(A[len(A)-1])
# leader=A[len(A)-1]
# i=A[len(A)-2]
# while(i>=0):
#     if(A[i]>leader):
#         print(A[i])
#         leader=A[i]
#         i-=1

arrSize=int(input())
arr=[]
for i in range(arrSize):
    arr.append(int(input()))
print(arr[len(arr)-1])
leader=arr[len(arr)-1]
for i in range(len(arr)-2,-1,-1):
    if(arr[i]>leader):
        print(arr[i])
        leader=arr[i]

File: ARX/array-leader.js
let fs = require("fs");
let data = fs.readFileSync("C:/Users/ASUS/OneDrive/Desktop/10x/input.txt", 'utf-8');
let idx = 0;
data = data.split('\n');

function readLine() {
    idx++;
    return data[idx - 1].trim();
}
let arrSize = parseInt(readLine());
let A = [];
for (let i = 0; i < arrSize; i++) {
    A.push(parseInt(readLine()));
}
console.log(A[A.length - 1]);
let leader = A[A.length - 1];
for (let i = A.length - 2; i >= 0; i--) {
    if (A[i] > leader) {
        console.log(A[i]);
        leader = A[i];
    }
}